### PR TITLE
Controller spec

### PR DESCRIPTION
**Pull request by gabrielcoder247**
### Controller spec
In this milestone.
I have added the following.

> Test to users_controller and posts_controller

### CODE REVIEW

File: test/controllers/posts_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  test 'should get index' do
+    get posts_index_url
+    assert_response :success
+  end
+
+  test 'should get show' do
+    get posts_show_url
+    assert_response :success
+  end
+end
```
Comment (Dipeshtwis) [inline test/controllers/posts_controller_test.rb:13]: - [ ] What you have written is the old method and it is not recommended for new application. `Using the older approach of mocking out the whole framework with ActionDispatch::ControllerTest is no longer recommended for new applications.` and hence in the new approach which is also recommended by Microverse is to  `Create a Request spec file for all your controllers` which I have show below how you can do or you can take reference from [here as well](https://relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec).

- [ ] when you have created the controller, it should have created `spec/requests/` directory with `type: request `  as shown in the example code. If not, you can create these folder manually and inside that you can create two file naming `users_spec.rb` and `posts_spec.rb`.

```ruby

# spec/requests/posts_spec.rb
require 'rails_helper'

 RSpec.describe "Posts", type: :request do
   ...
 end
```

and inside `Rspec.describe` you can write `describe` as shown below and inside `describe` you can write `it` block and test the controller.

```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
```

- [ ] Spec file include tests that ensures correct view template is rendered. Example : `expect(response).to render_template(:show)`

- [ ] Spec file include tests that ensures the response body includes the correct placeholder text. Example : `expect(response.body).to include("Posts#index")`

File: test/controllers/users_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  test 'should get index' do
+    get users_index_url
+    assert_response :success
+  end
+
+  test 'should get show' do
+    get users_show_url
+    assert_response :success
+  end
+end
```
Comment (Dipeshtwis) [inline test/controllers/users_controller_test.rb:13]: - [ ] Inside `users_spec.rb` file you can write like this as shown in the example code.
```ruby

# spec/requests/users_spec.rb
require 'rails_helper'

 RSpec.describe "Users", type: :request do
   ...
 end
```

and inside `Rspec.describe` you can write `describe` as shown below and inside `describe` you can write `it` block and test the controller.

```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
```

- [ ] Spec file include tests that ensures correct view template is rendered. Example : `expect(response).to render_template(:show)`

- [ ] Spec file include tests that ensures the response body includes the correct placeholder text. Example : `expect(response.body).to include("Users#index")`

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @gabrielcoder247, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**